src/testdir/test_viminfo.vim

```diff
@@ -315,6 +315,13 @@ func Test_cmdline_history_order()
   call delete('Xviminfo')
 endfunc
 
+func Force_write_viminfo()
+  redir => res
+  silent! wviminfo Xviminfo
+  redir END
+  call assert_match('W23:', res)
```

Comment:
instead of :redir, you can use `execute()`:

```suggestion
  let res = execute('wviminfo Xviminfo')
  call assert_match('W23:', res)
```

Reply:
I commited it along with the `silent!` since that allows the capturing of the warning